pkg/status/status.go

```diff
@@ -0,0 +1,193 @@
+	Services Services `json:"services"`
+
+	// Active Incidents and Maintenances
+	Incidents    Events `json:"incidents"`
+	Maintenances Events `json:"maintenances"`
+}
+
+// Get the status of global services (status of global or zones, no details of impacted services)
+func (s StatusPalStatus) GetStatusByZone() ([][]string, error) {
+	global := make([][]string, len(s.Services))
+	for index, svc := range s.Services {
+		state := svc.getIncidentType()
+		global[index] = []string{*svc.Name, state}
+	}
+	return global, nil
+}
+
+// A service can contains several child services
+// In our case:
+// - Parent services = Global and all the zones
+// - Child services = products available in a zone or globally
+type Service struct {
+	Id   *int    `json:"id,omitempty"`
+	Name *string `json:"name,omitempty"`
+
+	// The type of the current incident:
+	//  * `major` - A minor incident is currently taking place.
+	//  * `minor` - A major incident is currently taking place.
+	//  * `scheduled` - A scheduled maintenance is currently taking place.
+	//  * null - No incident is taking place.
+	IncidentType *string `json:"current_incident_type,omitempty"`
+
+	// Each product available in the zone
+	Children Services `json:"children,omitempty"`
+}
+
+func (s *Service) getIncidentType() string {
+	if s.IncidentType == nil {
+		return "operational"
+	}
+	switch *s.IncidentType {
+	case IncidentTypeScheduled:
+		return "scheduled maintenance"
+	default:
+		return fmt.Sprint(*s.IncidentType)
+	}
+}
+
+// Active Maintenance or Incident
+type Event struct {
+	Id    *int    `json:"id,omitempty"`
+	Title *string `json:"title,omitempty"`
+	// The time at which the incident/maintenance started(UTC).
+	StartsAt *string `json:"starts_at"`
+	// Type of current incident (major, minor, scheduled)
+
+	Type *string `json:"type"`
+
+	// Services impacted
+	ServiceIds []int `json:"service_ids"`
+}
+
+type Events []Event
+
+// Get Active Incidents or Maintenances with the full service name (Zone+Product)
+func (e Events) GetActiveEvents(services Services) ([][]string, error) {
+	var events EventsDetails
+
+	for _, event := range e {
+		// Get all the services impacted by the incident (name and id)
+		// Child and parent are all mixed, we need to rebuild the dependency
+		for _, impacted := range event.ServiceIds {
+			if services.IsParentService(impacted) {
+				continue
+			}
+			svcName, err := services.GetServiceNamebyId(impacted)
+			if err != nil {
+				return nil, err
+			}
+			started, err := time.Parse(dateLayout, *event.StartsAt)
+			if err != nil {
+				return nil, err
+			}
+			startTimeUTC := started.Format(time.RFC822)
+			eventDetails := []string{svcName, *event.Title}
+			if *event.Type == IncidentTypeScheduled {
+				eventDetails = append(eventDetails, "scheduled at "+startTimeUTC)
+				events = append(events, eventDetails)
+			} else {
+
+				eventDetails = append(eventDetails, fmt.Sprint(*event.Type), "since "+startTimeUTC)
```

Comment:
it looks like you forgot to remove `fmt.Sprint`, AFAICT it's not necessary

Reply:
https://github.com/exoscale/cli/pull/527/commits/ee45fd9ed94098dd3a92b678c61500e478804920